pysph/tools/gen_geom_surf.py

```diff
@@ -0,0 +1,121 @@
+'''
+Functions can be used to generate points to describe a given input
+mesh.
+
+Supported mesh formats: All file formats supported by meshio.
+(https://github.com/nschloe/meshio)
+'''
+
+import numpy as np
+import meshio
+from pysph.tools.geom_surf_points import get_surface_points,\
+                             get_surface_points_uniform
+
+
+class Mesh:
+    def __init__(self, file_name, file_type=None):
+        if file_type is None:
+            self.mesh = meshio.read(file_name)
+        else:
+            self.mesh = meshio.read(file_name, file_type)
+
+        self.cells = np.array([], dtype=int).reshape(0, 3)
+
+    def extract_connectivity_info(self):
+        cell_blocks = self.mesh.cells
+        for block in cell_blocks:
+            self.cells = np.concatenate((self.cells, block.data))
+
+        return self.cells
+
+    def extract_coordinates(self):
+        x, y, z = self.mesh.points.T
+        self.x, self.y, self.z = x, y, z
+
+        return x, y, z
+
+    def compute_normals(self):
+        n = self.cells.shape[0]
+        self.normals = np.zeros((n, 3))
+        points = self.mesh.points
+
+        for i in range(n):
+            idx = self.cells[i]
+            pts = np.array([points[idx[0]],
+                            points[idx[1]],
+                            points[idx[2]]])
+
+            normals = np.cross(pts[1] - pts[0], pts[2] - pts[0])
+            nrm = np.linalg.norm(normals)
+            self.normals[i] = normals/nrm
+
+        return self.normals
+
+
+def gen_surf_points(file_name, dx):
```

Comment:
This makes no sense, what is `gen_surf_points`?  How is it different from `get_surface_points`?

Reply:
The API has to be simple and easy to understand.  If you have two functions that are nearly similar and no real difference how is a user to know what to use?